application/classes/Ushahidi/Formatter/Media.php

```diff
@@ -55,11 +56,16 @@ protected function get_field_name($field)
 	}
 
 	protected function format_o_filename($value)
-	{
+	{		
+
 		if ($cdnBaseUrl = Kohana::$config->load('cdn.baseurl')) {
-			return $cdnBaseUrl . $value;
+			$url_path = explode("/", $value);
+			$filename = rawurlencode(array_pop($url_path));
+			array_push($url_path, $filename);
+
+			return $cdnBaseUrl . implode("/", $url_path);
 		} else {
-			return URL::site(Media::uri($this->get_relative_path() . $value), Request::current());
+			return URL::site((Media::uri($this->get_relative_path() . $value)), Request::current());
```

Comment:
It'd be worth adding a comment above this to explain that either URL::site or Media::uri is already encoding the url - so that the next person to look at this knows *why* we're only encoding in 1 case, not the other.